test/integration/factory.cc

```diff
@@ -1249,6 +1249,68 @@ TEST_F(FactoryTest, FilenameFuelURL)
   // Check model was spawned
   ASSERT_NE(nullptr, world->ModelByName("test_box"));
 }
+
+TEST_F(FactoryTest, worldWithFuelModels)
+{
+  this->Load("worlds/fuel_models.world", true);
+
+  // World
+  auto world = physics::get_world("default");
+  ASSERT_NE(nullptr, world);
+
+  // Wait for it to be spawned
+  int sleep = 0;
+  int maxSleep = 50;
+  while (!world->ModelByName("Radio") && sleep++ < maxSleep)
+  {
+    common::Time::MSleep(100);
+  }
+
+  // Check model was spawned
+  ASSERT_NE(nullptr, world->ModelByName("Radio"));
+}
+
+TEST_F(FactoryTest, FuelURIAsWorldArgument)
+{
+  this->Load(
+    "https://staging-fuel.ignitionrobotics.org/1.0/chapulina/worlds/Shapes",
```

Comment:
I don't think this test is working. For it to run for me, I had to remove the `#ifdef HAVE_IGNITION_FUEL_TOOLS` above (I believe it's a left over from [this PR](https://osrf-migration.github.io/gazebo-gh-pages/#!/osrf/gazebo/pull-requests/3139/page/1)).


I made a minor change to fix the test above and uploaded a test world to the OpenRobotics account that has models:

<details><summary>Diff</summary>
<p>

```.diff
diff --git a/test/integration/factory.cc b/test/integration/factory.cc
index f3c1afee3e..27e7903060 100644
--- a/test/integration/factory.cc
+++ b/test/integration/factory.cc
@@ -1222,7 +1222,6 @@ TEST_F(FactoryTest, FilenameModelDatabaseRelativePaths)
 }
 
 //////////////////////////////////////////////////
-#ifdef HAVE_IGNITION_FUEL_TOOLS
 TEST_F(FactoryTest, FilenameFuelURL)
 {
   this->Load("worlds/empty.world", true);
@@ -1233,7 +1232,7 @@ TEST_F(FactoryTest, FilenameFuelURL)
 
   msgs::Factory msg;
   msg.set_sdf_filename(
-      "https://api.ignitionfuel.org/1.0/chapulina/models/Test box");
+      "https://fuel.ignitionrobotics.org/1.0/chapulina/models/Test box");
 
   auto pub = this->node->Advertise<msgs::Factory>("~/factory");
   pub->Publish(msg);
@@ -1250,7 +1249,8 @@ TEST_F(FactoryTest, FilenameFuelURL)
   ASSERT_NE(nullptr, world->ModelByName("test_box"));
 }
 
-TEST_F(FactoryTest, worldWithFuelModels)
+//////////////////////////////////////////////////
+TEST_F(FactoryTest, WorldWithFuelModels)
 {
   this->Load("worlds/fuel_models.world", true);
 
@@ -1270,10 +1270,13 @@ TEST_F(FactoryTest, worldWithFuelModels)
   ASSERT_NE(nullptr, world->ModelByName("Radio"));
 }
 
+//////////////////////////////////////////////////
 TEST_F(FactoryTest, FuelURIAsWorldArgument)
 {
+  // ServerFixture::LoadArgs will split whitespaces, so we use a world name
+  // without spaces
   this->Load(
-    "https://staging-fuel.ignitionrobotics.org/1.0/chapulina/worlds/Shapes",
+    "https://fuel.ignitionrobotics.org/1.0/OpenRobotics/worlds/Test_shapes",
     true);
 
   // World
@@ -1311,7 +1314,6 @@ TEST_F(FactoryTest, FuelURIAsWorldArgument)
   // Check model was spawned
   ASSERT_NE(nullptr, world->ModelByName("sphere"));
 }
-#endif
 
 //////////////////////////////////////////////////
 TEST_P(FactoryTest, InvalidMeshInsertion)
```

</p>
</details>

 And then stopped at this error:

`[Err] [Server.cc:395] Could not open file[https://fuel.ignitionrobotics.org/1.0/OpenRobotics/worlds/Test_shapes]`

It looks like there are more `find_file` functions in `CommonIface` that need to be supported.